audit-base/src/main/java/com/wultra/core/audit/base/model/AuditRecord.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.wultra.core.audit.base.model;
+
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Audit record model class.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class AuditRecord {
+
+    private final Date timestamp;
```

Comment:
I am missing some:

- Correlation ID to be able to group records from the same process.
- Event type to simplify queries.

Reply:
I suggest we use the generic `param` map for this use case. As discussed, I will move these auditing parameters into a separate database table. The queries can be done on SQL level, e.g. `SELECT * FROM audit_log al INNER JOIN audit_param ap ON al.audit_log_id=ap.audit_log_id WHERE key = "correlation_id" and value = "..." `. Otherwise we would have to add new parameters into the `Audit` interface which would expand the interface quite a lot.

Reply:
Added in a separate pull request into this branch so that we can easily see the impact of these changes: https://github.com/wultra/lime-java-core/pull/48